### ISSUE TITLE
Rule Proposal: `@angular-eslint/prefer-signal-model`

### ISSUE THREAD

**Issue by splincode** [angular-eslint/angular-eslint#2766]
If I have this:

```ts
  public readonly tuiDropdownEnabled = input(false);
  public readonly tuiDropdownEnabledChange = output<boolean>();
```

then angular-eslint make it to

```ts
  public readonly tuiDropdownEnabled = model(false);
```

**Comment by hamfastgamgee** [comment]
I definitely could have used this when I was converting our apps to use input signals.  A few components had @Input and then @OutputEmitter with the same name and a Change suffix.  Running the angular migrations ended up producing input() and output() as expected.  But then inside the component there were updates being attempted from when it was an @Input to what was now input(), which of course is disallowed.  I switched the input() to model() without realizing the output() existed, and what happened was that the automatically generated output() inside model() hid the explicit one, so callbacks started silently failing with seemingly no warning.  In some cases just removing the explicit output() was right (because it really was just a standard ControlValueAccessor pattern but for whatever reason wasn't implemented that way) but in others I had to rename the output() and maintain it, because it was really being used as a callback and not for data binding.

In short: great idea!

**Comment by JamesHenry** [comment]
Sounds good, PRs welcome!

**Comment by mistrykaran91** [comment]
Hi! Is anybody working on it? Otherwise, I have tried adding this feature
